src/main/java/com/netgrif/application/engine/petrinet/service/ProcessRoleService.java

```diff
@@ -87,24 +92,27 @@ public void assignRolesToUser(String userId, Set<String> requestedRolesIds, Logg
         Set<ProcessRole> rolesRemovedFromUser = getRolesRemovedFromUser(userOldRoles, requestedRoles);
 
         String idOfPetriNetContainingRole = getPetriNetIdRoleBelongsTo(rolesNewToUser, rolesRemovedFromUser);
-
-        if (idOfPetriNetContainingRole == null)
+        if (!isGlobalFromFirstRole(rolesNewToUser) && !isGlobalFromFirstRole(rolesRemovedFromUser) && idOfPetriNetContainingRole == null) {
```

Comment:
Is assignment of multiple roles restricted to the same type? i.e. is it impossible to assign a net role together with a global role in a single operation? If this is allowed, the net role actions should still be run, no?